build-lib/util/licenser.js

```diff
@@ -21,22 +25,20 @@ Licenser.prototype.constructor = Licenser;
 
 Licenser.prototype.build = function () {
 
-  const files = this.inputPaths.reduce((fileAcc, dir) => {
-    const list = fs.readdirSync(dir).map(fileName => {
-      return path.join(dir, fileName);
-    });
-
-    return fileAcc.concat(list);
-  }, []).filter(fileName => {
-    return fileName.match(/^.+\.js$/);
-  });
-
-  files.forEach(fileName => {
-    const inputBuffer = fs.readFileSync(path.join(fileName));
+  this.inputPaths.forEach(dir => {
```

Comment:
This should be `const fileNames = [].concat(...this.inputPaths.map(recursiveReadDir))` followed by `fileNames.forEach(// write file)`. No need to conflate building up your data sets with side effects.